src/lambda_handler.py

```diff
@@ -1,89 +1,215 @@
 """Simple AWS Lambda handler to verify functionality."""
 
 # Standard Python Libraries
-from datetime import datetime, timezone
+import asyncio
+from datetime import datetime
+import gzip
+from io import BytesIO
+import json
 import logging
 import os
-from typing import Any, Optional, Union
+from typing import List, Optional, Tuple
+import urllib.parse
+import urllib.request
 
 # Third-Party Libraries
-import cowsay
-import cowsay.characters
-
-# cisagov Libraries
-from example import example_div
+from beanie import Document, init_beanie
+from boto3 import client as boto3_client
+from botocore.exceptions import ClientError
+from motor.motor_asyncio import AsyncIOMotorClient
 
 default_log_level = "INFO"
 logger = logging.getLogger()
 logger.setLevel(default_log_level)
 
+DEFAULT_NVD_URL = "https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-{year}.json.gz"
+NVD_FIRST_YEAR = 2002
+DEFAULT_NVD_COLLECTION = "cves"
+
+motor_client: AsyncIOMotorClient = None
+ssm_client: boto3_client = None
+
+
+class CVEDoc(Document):
+    """Python class to represent a CVE Document."""
+
+    id: str
+    cvss_score: Optional[float]
+    cvss_version: Optional[str]
+    severity: Optional[int]
+
+    class Settings:
+        """Optional settings."""
 
-def failed_task(result: dict[str, Any], error_msg: str) -> None:
-    """Update a given result because of a failure during processing."""
-    result["success"] = False
-    result["error_message"] = error_msg
+        name = DEFAULT_NVD_COLLECTION
+        validate_on_save = True
 
+    async def save(self, *args, **kwargs):
+        """Save the document to the database."""
+        # Calculate severity based on CVSS version and score.
 
-def task_default(event):
-    """Provide a result if no valid task was provided."""
-    result = {}
-    error_msg = 'Provided task "%s" is not supported.'
+        if self.cvss_version == "2.0":
+            if self.cvss_score == 10:
+                self.severity = 4
+            elif self.cvss_score >= 7.0:
+                self.severity = 3
+            elif self.cvss_score >= 4.0:
+                self.severity = 2
+            else:
+                self.severity = 1
+        elif self.cvss_version in ["3.0", "3.1"]:
+            if self.cvss_score >= 9.0:
+                self.severity = 4
+            elif self.cvss_score >= 7.0:
+                self.severity = 3
+            elif self.cvss_score >= 4.0:
+                self.severity = 2
+            else:
+                self.severity = 1
+        await super().save(*args, **kwargs)
 
-    task = event.get("task", None)
-    logging.error(error_msg, task)
-    failed_task(result, error_msg % task)
 
-    return result
+async def process_nvd(cve) -> str:
+    """Add the provided CVE to the database and return its id."""
+    # Fill fields for the CVE Documents from the given JSON files
```

Comment:
Remove unnecessary capitalization:
```suggestion
    # Fill fields for the CVE documents from the given JSON files
```

Reply:
@dav3r I capitalized this because it was the name of the Beanie ODM class Document. See here https://beanie-odm.dev/tutorial/defining-a-document/ 

That being said they do call the section "Defining a _document_" with no capitalization so I am fine either way but still thought it was worth mentioning.

Reply:
Understood, but in this comment, I still think `documents` reads more cleanly than `Documents`.  Go with whatever your ❤️  tells you and I will support it.

Reply:
https://www.youtube.com/watch?v=yCC_b5WHLX0

Reply:
https://www.youtube.com/watch?v=Ud6sU3AclT4

Reply:
@jsf9k and @dav3r I feel Rick Rolled having clicked on both of those links haha

Reply:
@dav3r I addressed this here [Line 75](https://github.com/cisagov/cyhy-nvdsync-lambda/pull/2/commits/e6397c680698de8835b0629e200b28fe2829df5b#diff-4655711b74e9f79a2566fdbac7451f4dcc265a12d3ad743e519353283898a155R75) I went with my ❤️

Reply:
Thanks for taking care of this @michaelsaki.  I do have to point out that https://github.com/cisagov/cyhy-nvdsync-lambda/pull/2/commits/e6397c680698de8835b0629e200b28fe2829df5b is a big, gnarly commit that does a bunch of different things.  In the future, please try to keep your commits smaller and more focused on a single thing.  It makes them easier to review and understand.

Reply:
@dav3r You are right I did think that after I had made the commit. I will do better next time.